packages/binding-http/src/http-server.ts

```diff
@@ -570,7 +570,16 @@ export default class HttpServer implements ProtocolServer {
 
                 return acceptValue;
             })
-            .filter((acceptValue) => contentSerdes.isSupported(acceptValue));
+            .filter((acceptValue) => contentSerdes.isSupported(acceptValue))
+            .sort((a, b) => {
+                // weight function last places weight more than first: application/td+json > application/json > text/html
+                const aWeight =
+                    ["text/html", "application/json", "application/td+json"].findIndex((value) => value === a) ?? 0;
+                const bWeight =
+                    ["text/html", "application/json", "application/td+json"].findIndex((value) => value === b) ?? 0;
```

Comment:
I wonder whether the fallback should **not** be `0` but maybe `-1`

Let's assume some fancy value comes in like `application/bla` besides `text/html`. Both would result in index 0 while I think we should favor `text/html` ...

Reply:
I forgot that `findIndex` returns already -1 if the element is not found (not undefined), therefore `??` is pointless. I removed it and added a relevant test.